src/java.base/share/classes/java/nio/channels/FileChannel.java

```diff
@@ -554,7 +554,10 @@ public final long write(ByteBuffer[] srcs) throws IOException {
      * actually done is system-dependent and is therefore unspecified.
      *
      * <p> This method is only guaranteed to force changes that were made to
-     * this channel's file via the methods defined in this class.  It may or
+     * this channel's file via the methods defined in this class or, if this
+     * channel was derived from a {@link FileOutputStream} or
+     * {@link RandomAccessFile} using the {@code getChannel} method,
+     * via the methods defined in those classes.  It may or
```

Comment:
This looks okay, I just note that the existing javadoc uses "obtain" rather than "derive". An alternative to consider is "or the methods defined by FOS or RAF when the channel was obtained with the getChannel method".

Reply:
So changed.